src/document_basic.c

```diff
@@ -65,7 +68,10 @@ void Document_MakeStringsOwner(Document *d) {
 
   for (size_t ii = 0; ii < d->numFields; ++ii) {
     DocumentField *f = d->fields + ii;
-    f->name = rm_strdup(f->name);
+    if (f->path != f->name) {
+      f->name = rm_strdup(f->name);
+    }
+    f->path = rm_strdup(f->path);
```

Comment:
I think it would be better to create a struct that will handle fields name/paths. I see it's handled all around...

Reply:
I do not see why adding a struct is excessive, it does not have to be a pointer, we can still put it as part of the bigger struct. But we can create an api that handles the free, assigments, ...
I see all over the code that we check if path == name { free ..}, I think we can create a unit for freeing and populate this struct for example ...